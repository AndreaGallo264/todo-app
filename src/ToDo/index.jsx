import React from "react";
import "./style.css";

const ToDo = ({ todo, handleToggle }) => {
  return (
    <>
      <div
        className={todo.complete ? "strike" : ""}
        onClick={() => {
          handleToggle(todo.id);
        }}
      >
        {todo.task}
      </div>
      <div>{todo.dueDate}</div>
    </>
  );
};

export default ToDo;
