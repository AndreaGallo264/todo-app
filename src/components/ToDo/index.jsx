import React from "react";
import { useDispatch } from "react-redux";
import { ListItem, ListItemIcon, ListItemText } from "@material-ui/core";
import CheckCircleOutlineIcon from "@material-ui/icons/CheckCircleOutline";
import CancelIcon from "@material-ui/icons/Cancel";
import { changeTaskstate } from "../../store/actions";
import { formatDate } from "../Utils/formatDate";

const ToDo = ({ todo }) => {
  const formatedDate = formatDate(todo.dueDate);
  const dispatch = useDispatch();

  const renderIcon = (completed) => {
    if (completed) {
      return <CheckCircleOutlineIcon color="primary" />;
    } else {
      return <CancelIcon color="secondary" />;
    }
  };
  return (
    <ListItem selected={todo.complete} button>
      <ListItemIcon>{renderIcon(todo.complete)}</ListItemIcon>
      <ListItemText
        onClick={() => {
          dispatch(changeTaskstate(todo.id));
        }}
        primary={todo.task}
        secondary={`Due date: ${formatedDate}`}
      />
    </ListItem>
  );
};

export default ToDo;
