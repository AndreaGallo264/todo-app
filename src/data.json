[{
  "id": 1,
  "task": "Give dog a bath",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 2,
  "task": "Do laundry",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 3,
  "task": "Vacuum floor",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 4,
  "task": "Feed cat",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 5,
  "task": "Change light bulbs",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 6,
  "task": "Go to Store",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 7,
  "task": "Fill gas tank",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 8,
  "task": "Change linens",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 9,
  "task": "Rake leaves",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 10,
  "task": "Bake Cookies",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 11,
  "task": "Take nap",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 12,
  "task": "Read book",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 13,
  "task": "Exercise",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 14,
  "task": "Give dog a bath",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 15,
  "task": "Do laundry",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 16,
  "task": "Vacuum floor",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 17,
  "task": "Feed cat",
  "complete": true,
  "dueDate": "2021-05-08"
}, {
  "id": 18,
  "task": "Change light bulbs",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 19,
  "task": "Go to Store",
  "complete": false,
  "dueDate": "2021-05-08"
}, {
  "id": 20,
  "task": "Fill gas tank",
  "complete": false,
  "dueDate": "2021-05-08"
}]