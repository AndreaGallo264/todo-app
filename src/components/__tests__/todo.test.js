import React from "react";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import data from "../../data.json";
import ToDo from "../ToDo";

describe("ToDo component", () => {
  test("Matches snapshot", () => {
    const mockStore = configureMockStore();
    const store = mockStore({ list: data, lastId: data.length });
    const tree = renderer
      .create(
        <Provider store={store}>
          <ToDo
            todo={{
              id: 1,
              task: "Give dog a bath",
              complete: true,
              dueDate: "2021-05-08",
            }}
          />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
