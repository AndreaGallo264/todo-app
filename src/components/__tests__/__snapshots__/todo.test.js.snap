// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDo component Matches snapshot 1`] = `
<div
  aria-disabled={false}
  className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button Mui-selected"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="MuiListItemIcon-root"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M16.59 7.58L10 14.17l-3.59-3.58L5 12l5 5 8-8zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
    </svg>
  </div>
  <div
    className="MuiListItemText-root MuiListItemText-multiline"
    onClick={[Function]}
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
    >
      Give dog a bath
    </span>
    <p
      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
    >
      Due date: 2021-05-08
    </p>
  </div>
</div>
`;
